File: src/intermezzo/input.cpp
#include <cstdint>
#include <iostream>
#include <vector>

void copy_odd(const std::vector<int32_t> & src, std::vector<int32_t> & dst) {
    for (const int32_t n : src) {
        if (n & 0x1) {
            dst.push_back(n);
        }
    }
}

int main(const int, const char * []) {
    std::vector<int32_t> v{1, 2, 3, 4, 5};
    copy_odd(v, v);
    for (const int32_t n : v) {
        std::cout << n << std::endl;
    }
}

File: src/intermezzo/input.rs
#[allow(dead_code)]
#[allow(clippy::ptr_arg)]
fn copy_odd(src: &Vec<i32>, dst: &mut Vec<i32>) {
    let odd = src.iter().filter(|n| *n & 0x1 != 0);
    dst.extend(odd);
}

fn main() {
    /*
        let mut v = vec![1, 2, 3, 4, 5];
        copy_odd(&v, &mut v);
        for n in &v {
            println!("{n}");
        }
    */
}
